src/parser/meta.rs

```diff
@@ -1,35 +1,20 @@
 use nom::{
     branch::alt,
     bytes::streaming::{tag, take, take_while1},
-    character::{
-        complete::digit1,
-        streaming::{crlf, space0, space1},
-    },
-    combinator::{map, map_res, value},
+    character::streaming::{crlf, space1},
+    combinator::{map, value},
     error::ErrorKind::Fail,
     multi::many0,
-    sequence::{pair, preceded, terminated, tuple},
+    sequence::tuple,
     IResult,
 };
 
+use std::num::NonZero;
+
 use super::{parse_u32, ErrorKind, MetaValue, Response, Status, Value};
 use crate::Error;
 
-// TODO: remove this later in favour of individual parse_meta_*_status methods
```

Comment:
Removed this because using one general function doesn't work in practice, as expected.  There is some overlap between response codes in terms of differing interpretations and each op needs to be able to trigger the right logic when encountering a response code.